routes/soldiers.js

```diff
@@ -0,0 +1,31 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, soldierData, done) {
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  const parsedUrlLength = req.url.split("/").length
+  const soldierId = req.url.split("/")[2];
+  if (parsedUrlLength > 3) {
+    return done(new Error())
```

Comment:
Make sure to return a message in the error
`new Error("message that tells the user what he did wrong")`

Reply:
Or more like Pavel said, return an error with the correct status.
`badRequest` usually comes with a general description of what's bad about it